src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -5,8 +5,7 @@
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
-
+public interface MachineService<T> {
     /**
```

Comment:
Parameterize MachineProducer in such a way that any implementation NOT using Machine or its subtype could NOT be created.
For example, this code should not compile:

public class StringProducer implements MachineProducer\\\<String>\\\ { }